internal/dag/gatewayapi_processor.go

```diff
@@ -1102,6 +1103,33 @@ func (p *GatewayAPIProcessor) resolveRouteRefs(route any, routeAccessor *status.
 	}
 }
 
+func validateTimeout(timeout *gatewayapi_v1.HTTPRouteTimeouts) (*RouteTimeoutPolicy, *RetryPolicy, error) {
+	if timeout == nil || (timeout.BackendRequest == nil && timeout.Request == nil) {
+		return nil, nil, nil
+	}
+
+	tr := &contour_api_v1.TimeoutPolicy{}
+	if timeout.Request != nil {
+		tr.Response = string(*timeout.Request)
+	}
+
+	rtp, _, err := timeoutPolicy(tr, 0)
+	if err != nil {
+		return nil, nil, fmt.Errorf("parse \"Request\" is failed: %s", err)
+	}
+
+	br := &contour_api_v1.RetryPolicy{}
+	if timeout.BackendRequest != nil {
+		br.PerTryTimeout = string(*timeout.BackendRequest)
+	}
+
+	rp := retryPolicy(br)
+	if rp.PerTryTimeout.Duration() > rtp.ResponseTimeout.Duration() {
+		rtp.ResponseTimeout = rp.PerTryTimeout
+	}
+	return &rtp, rp, nil
```

Comment:
I _think_ this implementation is fine for now while Gateway API does not yet support retries, but some observations for reviewers/others on how this is working:

- `retryPolicy(...)` enables a single retry (i.e. up to two total backend requests) by default. However, since for Gateway API we don't specify any retriable conditions, I don't think Envoy will actually retry any requests
- given that the conformance tests pass, I _assume_ that the per-try timeout is still taking effect for that first request
- I think it'd be good to set `br.NumRetries` to -1 to explicitly disable retries for now, assuming that the per-try timeout still takes effect on the first backend request in that case. It's a little bit more self-documenting in the Envoy config that way.
- an alternate way we could implement this now is just to set the route "response" timeout to the smaller of the Request or BackendRequest timeout. It might be a little clearer/more obvious for now, but once Gateway API supports retries, then this code will become necessary again. So, I'm fine keeping the implementation as-is given the above notes.

Reply:
I did some testing, and it looks like the current code actually does enable retries here (which is a behavior change), since we default to retrying 5xx codes if no other retry conditions are set, and Envoy defaults to a single retry if no other retry count is specified.

The other option would be to not enable an automatic retry, in which case it doesn't really make sense to support `timeouts.backendRequest` right now since without retries it's the same as `timeouts.request`. Once retries are added to the Gateway API spec, then we can circle back and implement them properly, along with `timeouts.backendRequest`.

I'm leaning towards the latter option, to keep things simple and avoid the behavior change of automatically adding a retry, which would likely change again when Gateway API gets proper retry support.

cc @sunjayBhatia would appreciate your thoughts here as well.

Reply:
ping @sunjayBhatia

Reply:
+1 for not changing retry behavior and keeping things as is until there is an upstream retry API

if we can set a backend request timeout without enabling retries at all that would be ideal

cases to consider (and have tests for):
- both timeout values are set
- only `request` timeout is set: seems straightforward
- only `backendRequest` is set

Reply:
if we want to explicitly say we don't support the `backendRequest` timeout we can do so with status conditions or just fail the conformance test for now